src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -339,48 +352,18 @@ public Task<ODataBatchOperationResponseMessage> CreateOperationResponseMessageAs
 #endif
 
         /// <summary>Flushes the write buffer to the underlying stream.</summary>
-        public void Flush()
-        {
-            this.VerifyCanFlush(true);
-
-            // make sure we switch to state FatalExceptionThrown if an exception is thrown during flushing.
-            try
-            {
-                this.rawOutputContext.Flush();
-            }
-            catch
-            {
-                this.SetState(BatchWriterState.Error);
-                throw;
-            }
-        }
+        public abstract void Flush();
 
 #if PORTABLELIB
         /// <summary>Flushes the write buffer to the underlying stream asynchronously.</summary>
         /// <returns>A task instance that represents the asynchronous operation.</returns>
-        public Task FlushAsync()
-        {
-            this.VerifyCanFlush(false);
-
-            // make sure we switch to state FatalExceptionThrown if an exception is thrown during flushing.
-            return this.rawOutputContext.FlushAsync().FollowOnFaultWith(t => this.SetState(BatchWriterState.Error));
-        }
+        public abstract Task FlushAsync();
 #endif
 
         /// <summary>
         /// This method is called to notify that the content stream for a batch operation has been requested.
         /// </summary>
-        void IODataBatchOperationListener.BatchOperationContentStreamRequested()
-        {
-            // Write any pending data and flush the batch writer to the async buffered stream
-            this.StartBatchOperationContent();
-
-            // Flush the async buffered stream to the underlying message stream (if there's any)
-            this.rawOutputContext.FlushBuffers();
-
-            // Dispose the batch writer (since we are now writing the operation content) and set the corresponding state.
-            this.DisposeBatchWriterAndSetContentStreamRequestedState();
-        }
+        public abstract void BatchOperationContentStreamRequested();

```

Comment:
>BatchOperationContentStreamRequested [](start = 29, length = 36)

Since BatchOperationContentStreamRequested is already defined in IODataBatchOperationListener, which ODataBatchWriter is derived from, do you need this line (or the next one?)  That is, the derived class has to implement it anyway, because it's defined in the interface that the (now abstract base) class ODataBatchWriter implements.

Reply:
But w/o the abstract declaration, I got an error (from VS):
ODataBatchWriter.cs(23,46,23,74): error CS0535: 'ODataBatchWriter' does not implement interface member 'IODataBatchOperationListener.BatchOperationContentStreamRequested()'

Does that work for you?

---
In reply to: [140938082](https://github.com/OData/odata.net/pull/960#discussion_r140938082) [](ancestors = 140938082)